osu.Game/Screens/Play/HUD/QuitButton.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input;
+using osu.Framework.MathUtils;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using OpenTK;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    public class QuitButton : FillFlowContainer
+    {
+        public override bool ReceiveMouseInputAt(Vector2 screenSpacePos) => button.ReceiveMouseInputAt(screenSpacePos);
+
+        private readonly Button button;
+
+        public Action Action
+        {
+            set => button.Action = value;
+        }
+
+        private readonly OsuSpriteText text;
+
+        public QuitButton()
+        {
+            Direction = FillDirection.Horizontal;
+            Spacing = new Vector2(20, 0);
+            Margin = new MarginPadding(10);
+            Children = new Drawable[]
+            {
+                text = new OsuSpriteText
+                {
+                    Text = "hold for menu",
+                    Font = @"Exo2.0-Bold",
+                    Anchor = Anchor.CentreLeft,
+                    Origin = Anchor.CentreLeft
+                },
+                button = new Button()
+            };
+            AutoSizeAxes = Axes.Both;
+        }
+
+        protected override void LoadComplete()
+        {
+            text.FadeInFromZero(500, Easing.OutQuint).Delay(1500).FadeOut(500, Easing.OutQuint);
+            base.LoadComplete();
+        }
+
+        protected override bool OnHover(InputState state)
+        {
+            text.FadeIn(500, Easing.OutQuint);
+            return true;
+        }
+
+        protected override void OnHoverLost(InputState state)
+        {
+            text.FadeOut(500, Easing.OutQuint);
+            base.OnHoverLost(state);
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            float adjust = Vector2.Distance(GetContainingInputManager().CurrentState.Mouse.NativeState.Position, button.ScreenSpaceDrawQuad.Centre) / 200;
+            double elapsed = MathHelper.Clamp(Clock.ElapsedFrameTime, 0, 1000);
+
+            bool stayVisible = text.Alpha > 0 || button.Progress.Value > 0 || IsHovered;
+
+            Alpha = stayVisible ? 1 : Interpolation.ValueAt(elapsed, Alpha, MathHelper.Clamp(1 - adjust, 0.04f, 1), 0, 200, Easing.OutQuint);
```

Comment:
We'll want a debounce, otherwise it's going to trigger readily for 4:3 players.

Reply:
Also this affects the alpha while the pause screen is active, looks pretty weird.